design/dsl/endpoint_test.go

```diff
@@ -3,114 +3,134 @@ package dsl_test
 import (
 	"testing"
 
+	"reflect"
+
 	"github.com/goadesign/goa/design"
 	. "github.com/goadesign/goa/design/dsl"
 	"github.com/goadesign/goa/eval"
 )
 
 func TestEndpoint(t *testing.T) {
 	cases := map[string]struct {
-		Expr        eval.Expression
-		Invocations int
-		Description string
-		Doc         string
-		DocURL      string
-		Name        string
-		RequestAttr map[string]design.Primitive
+		Expr   eval.Expression
+		DSL    func()
+		Assert map[string]func(t *testing.T, s *design.EndpointExpr)
```

Comment:
:) Is that really necessary? Each Assert is free to check whatever they need to. That is the Assert function itself could contain the code to check both endpoints. This would keep the data structure nice and simple and the iteration code also simple was still allowing all the needed flexibility.